doc/snmp/extension-to-physical-entity-mib.md

```diff
@@ -0,0 +1,149 @@
+	         |              |-- power-mon/ VOLTAGE  (Voltage sensor)
+	         |--Ethernet x/y … cable (Port module)
+                            |--DOM Temperature Sensor for Ethernet(x)  (Temperature sensor)
+                            |--DOM Voltage Sensor for Ethernet(x)  (Voltage sensor)
+                            |--DOM RX Power Sensor for Ethernet(x)/(y) (Power sensor)
+                            |--DOM TX Bias Sensor for Ethernet(x)/(y) (Bias sensor)
+
+
+## 4. The data source of the MIB entries
+
+Thermalctl daemon, Xcvrd, psud, are collecting physical device info to state DB, now we have PSU_INFO tale, FAN_INFO table, and TEMPERATURE_INFO table which can provide information for MIB entries. 
+
+Thermal sensors MIB info will come from TEMPERATURE_INFO, FAN_INFO will feed to FAN MIB entries and PSU_INFO will be the source of the PSU related entries.
+
+The current already implemented cable and cable DOM sensors getting data from tables(TRANSCEIVER_INFO and TRANSCEIVER_DOM_SENSOR) which maintained by xcvrd.
+
+### 4.1 Adding more data to state DB
+
+As mentioned in Section 3, currently lack the implementation of "entPhysicalContainedIn" object, and also there is no such kind of info stored in the state DB. A "contained_in_device" device will be added to the current tables, e.g., extended TEMPERATURE_INFO with the new field:
+
+	; Defines information for a thermal object
+	key                     = TEMPERATURE_INFO|object_name   ; name of the thermal object(CPU, ASIC, optical modules...)
+	; field                 = value
+	temperature             = FLOAT                          ; current temperature value
+	timestamp               = STRING                         ; timestamp for the temperature fetched
+	high_threshold          = FLOAT                          ; temperature high threshold
+	critical_high_threshold = FLOAT                          ; temperature critical high threshold
+	low_threshold           = FLOAT                          ; temperature low threshold
+	critical_low_threshold  = FLOAT                          ; temperature critical low threshold
+	warning_status          = BOOLEAN                        ; temperature warning status
+	contained_in_device     = STRING                         ; name of the device which contains this sensor
+
+To have this kind info available we need to extend the current platform API, can add a new function "get_contained_in_device_name" to DeviceBase class, this function will return the name of the device which it is contained in. This new field will be populated to state DB by related PMON daemons.
+
+	class DeviceBase(object):
+	    """
+	    Abstract base class for interfacing with a generic type of platform
+	    peripheral device
+	    """
+
+	    def get_contained_in_device_name(self):
+		"""
+		Retrieves The name of the device which contains this component
+		Returns:
+		    string: The name of the device
+		"""
+		raise NotImplementedError
+
+## 5. Entity MIB extension test
```

Comment:
will a new deamon to be added to put those data in the state db?

Reply:
no new daemon needed, only need to extend current daemons to collect more data to DB which is required for the MIB